src/Adapters/AzureManagementAdapter.ts

```diff
@@ -567,4 +571,94 @@ export default class AzureManagementAdapter implements IAzureManagementAdapter {
             return new AzureResourcesData([newRoleAssignment]);
         }, 'azureManagement');
     }
+
+    async getTimeSeriesConnectionInformation(adtInstanceIdentifier: {
+        id?: string;
+        hostName?: string;
+    }) {
+        // either the resource object or host url of the ADT instance
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+
+        return await adapterMethodSandbox.safelyFetchData(async (token) => {
+            let adtInstance: IAzureResource;
+            if (adtInstanceIdentifier.hostName) {
+                const digitalTwinInstances = await this.getResourcesByPermissions(
+                    {
+                        getResourcesParams: {
+                            resourceType: AzureResourceTypes.DigitalTwinInstance
+                        },
+                        requiredAccessRoles: RequiredAccessRoleGroupForADTInstance
+                    }
+                );
+                const result = digitalTwinInstances.result.data;
+                adtInstance = result.find(
+                    (d) =>
+                        d.properties.hostName === adtInstanceIdentifier.hostName
+                );
+            } else if (adtInstanceIdentifier.id) {
+                const digitalTwinInstance = await this.getResourceById(
+                    adtInstanceIdentifier.id
+                );
+                adtInstance = digitalTwinInstance.result.data;
+            }
+
+            try {
```

Comment:
do we need to confirm that we got back an ADT Instance before proceeding?

Reply:
nope, since it is in try catch block, if instance not exist it will fall into catch and adapter will resolve null (just added that).